File: Go/company_db/backend/models/company.go
package models

import (
	"time"

	"gorm.io/datatypes"
	"gorm.io/gorm"
)

// Company モデル
type Company struct {
	ID           string         `gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	Name         string         `gorm:"type:varchar(255);not null"`
	URL          datatypes.JSON `gorm:"type:jsonb"` // JSONB で保存
	PhoneNumber  string         `gorm:"type:varchar(20)"`
	PostalCodeID int            `gorm:"type:integer"`
	PostalCode   PostalCode     `gorm:"foreignKey:PostalCodeID;references:ID"`
	Address      string         `gorm:"type:varchar(255)"`
	CreatedAt    time.Time      `gorm:"default:now()"`
	UpdatedAt    time.Time      `gorm:"default:now()"`
	DeletedAt    gorm.DeletedAt `gorm:"index"`

	// 多対多のリレーション
	Tags []Tag `gorm:"many2many:tag_companies;"`
}

File: Python/useRedis/app/models/company.py
from app.db.database import Base
from sqlalchemy import TIMESTAMP, Column, ForeignKey, Integer, String, func


class Company(Base):
    __tablename__ = "companies"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    postal_code_id = Column(Integer, ForeignKey("postal_codes.id"))
    prefecture_id = Column(Integer, ForeignKey("prefectures.id"))
    address = Column(String)
    contact_name = Column(String)
    phone = Column(String)
    created_at = Column(TIMESTAMP, server_default=func.now())
    updated_at = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now())
